qemu/qemu.go

```diff
@@ -2103,6 +2103,50 @@ type Kernel struct {
 	Params string
 }
 
+// FwCfg allows QEMU to pass entries to the guest
+// File and Str are mutually exclusive
+type FwCfg struct {
+	Name string
+	File string
+	Str  string
+}
+
+// Valid returns true if the FwCfg structure is valid and complete.
+func (fwcfg FwCfg) Valid() bool {
+	if fwcfg.Name == "" {
+		return false
+	}
+
+	if fwcfg.File != "" && fwcfg.Str != "" {
+		return false
+	}
+
```

Comment:
now you need to check that both are not empty (`-fw_cfg` needs a file or a string)

```go
if fwcfg.File == "" && fwcfg.Str == "" {
    return false
}
```